acceptance/upgrade/upgrade_bound_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright © 2021 - 2023 SUSE LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upgrade_test
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/epinio/epinio/acceptance/helpers/catalog"
+	"github.com/epinio/epinio/acceptance/helpers/epinio"
+	"github.com/epinio/epinio/acceptance/testenv"
+
+	. "github.com/epinio/epinio/acceptance/helpers/matchers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("<Upgrade2> Epinio upgrade with bound app and services", func() {
+	var (
+		namespace string // Namespace created before upgrade
+		appName   string // Application created before upgrade
+		service   string // Service created after upgrade
+
+		epinioHelper epinio.Epinio
+	)
+
+	BeforeEach(func() {
+		epinioHelper = epinio.NewEpinioHelper(testenv.EpinioBinaryPath())
+
+		// Before upgrade ...
+		namespace = catalog.NewNamespaceName()
+		env.SetupAndTargetNamespace(namespace)
+		appName = catalog.NewAppName()
+		service = catalog.NewServiceName()
+	})
+
+	AfterEach(func() {
+		env.DeleteNamespace(namespace)
+	})
```

Comment:
You can use a `DeferCleanup`, so it's clearer that this is needed to "cleanup" after the setup:
```suggestion
	BeforeEach(func() {
		epinioHelper = epinio.NewEpinioHelper(testenv.EpinioBinaryPath())

		// Before upgrade ...
		namespace = catalog.NewNamespaceName()
		env.SetupAndTargetNamespace(namespace)
		appName = catalog.NewAppName()
		service = catalog.NewServiceName()

		DeferCleanup(func() {
			env.DeleteNamespace(namespace)
		})	
	})
```